### PR TITLE
Upgrade to django 4.2

### PR DESCRIPTION
**Pull request by davidu1975**
Upgrade to django 4.2

### Workflow

- [x] Ticket exists in Jira https://uktrade.atlassian.net/browse/KLS-1737
- [x] Jira ticket has the correct status.
- [x] A clear/description pull request messaged added.

### Merging

- [x] This PR can be merged by reviewers. (If unticked, please leave for the author to merge)

### CODE REVIEW

File: company/migrations/0108_alter_companyuser_company.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 4.1.13 on 2024-01-15 11:12
+
+from django.db import migrations, models
```
Comment (sebs-code) [inline company/migrations/0108_alter_companyuser_company.py:3]: Why are there migrations in a Django bump?

Comment (bobby-didcoding) [reply company/migrations/0108_alter_companyuser_company.py:3]: Hi @sebs-code, somewhere along the line the models had been adjusted without migration files being created. They don't need to be included in this release but will need to be added at some point.

Comment (davidu1975) [reply company/migrations/0108_alter_companyuser_company.py:3]: Thanks @bobby-didcoding

Comment (sebs-code) [reply company/migrations/0108_alter_companyuser_company.py:3]: They might as well go in now... if they have to at some point!

**Review by sebs-code — APPROVED** [review]

**Pull request merged**